src/main/java/com/mewna/catnip/shard/manager/DefaultShardManager.java

```diff
@@ -90,8 +91,9 @@ private void loadShards(final int count) {
         catnip.logAdapter().info("Booting {} shards", count);
         // Deploy verticles
         for(int id = 0; id < count; id++) {
+            // because each shard has its own presence, so no global presence on catnip class
             //noinspection TypeMayBeWeakened
-            final CatnipShard shard = new CatnipShard(catnip, id, count);
+            final CatnipShard shard = new CatnipShard(catnip, id, count, ((CatnipImpl) catnip).presence());
```

Comment:
Each shard may have its own presence, but the presence being passed to the global catnip object is the INITIAL presence, which will be the same for all shards spawned under the same catnip instance; this shouldn't require casting to `CatnipImpl` I feel. It might be more logical to expose this on the catnip instance as something like `Catnip#initialPresence` instead of just `presence`

Reply:
It might also make sense to have `Catnip#presence(int)` to get it for a specific shard

Reply:
yeah i was planning on exposing that as `Catnip#initialPresence` earlier but dropped that change. also, as of rn, you can't access `CatnipShard` instances which is why i have the custom string format as well as no shard-specific methods

Reply:
done this but i haven't added specific methods to get the presence of a shard, so not resolved properly yet